install.sh

```diff
@@ -36,20 +51,73 @@ fi
 read -p "Do you want to run the game when the installation ends? (y/N): " run_game
 
 
+install_docker_official() {
+    echo "Installing Docker using the official installation script..."
+    curl -fsSL https://get.docker.com -o get-docker.sh
+    sudo sh get-docker.sh
+    sudo systemctl start docker
+    sudo systemctl enable docker
+}
+
 if [[ $(uname) == "Linux" ]]; then
-    echo "Setting up the environment."
-    sudo apt update
-    sudo apt install docker.io git curl -y
-    sudo curl -L "https://github.com/docker/compose/releases/download/v2.20.3/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose
-    sudo chmod +x /usr/local/bin/docker-compose
-elif [[$(uname) == "Darwin" ]]; then
-    echo "Installing homebrew"
-    sudo curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh
+    echo "Setting up the environment for Linux."
+    
+    # Detect the package manager
+    if command -v apt &> /dev/null || command -v dnf &> /dev/null || command -v yum &> /dev/null; then
+        if command -v apt &> /dev/null; then
+            PKG_MANAGER="apt"
+            INSTALL_CMD="sudo apt update && sudo apt install -y"
+        elif command -v dnf &> /dev/null; then
+            PKG_MANAGER="dnf"
+            INSTALL_CMD="sudo dnf install -y"
```

Comment:
with `apt` using update before installing, maybe dnf should've used the `--best` parameter or `sudo dnf update` to get the latest package